pootle/core/views.py

```diff
@@ -359,3 +373,142 @@ def page_not_found(request):
+        return super(PootleDetailView, self).dispatch(
+            self.decorate_request(request), *args, **kwargs)
+
+
+class PootleBrowseView(PootleDetailView):
+    template_name = 'browser/index.html'
+
+    @property
+    def stats(self):
+        return self.object.get_stats()
+
+    @property
+    def has_vfolders(self):
+        return False
+
+    @property
+    def sidebar_announcements(self):
+        return get_sidebar_announcements_context(
+            self.request,
+            (self.object, ))
+
+    def get_context_data(self, *args, **kwargs):
+        filters = {}
+        if self.has_vfolders:
+            filters['sort'] = 'priority'
+
+        url_action_continue = self.object.get_translate_url(
+            state='incomplete',
+            **filters)
+        url_action_fixcritical = self.object.get_critical_url(
+            **filters)
+        url_action_review = self.object.get_translate_url(
+            state='suggestions',
+            **filters)
+        url_action_view_all = self.object.get_translate_url(state='all')
+        ctx, cookie_data = self.sidebar_announcements
+        ctx.update(
+            super(PootleBrowseView,
+                  self).get_context_data(*args, **kwargs))
+        ctx.update(
+            {'page': 'browse',
+             # 'parent': get_parent(self.object),
+             'pootle_path': self.object.pootle_path,
+             'stats': jsonify(self.stats),
+             'translation_states': get_translation_states(self.object),
+             'check_categories': get_qualitycheck_schema(self.object),
+             'url_action_continue': url_action_continue,
+             'url_action_fixcritical': url_action_fixcritical,
+             'url_action_review': url_action_review,
+             'url_action_view_all': url_action_view_all,
+             'browser_extends': 'translation_projects/base.html'})
+        return ctx
+
+
+class PootleTranslateView(PootleDetailView):
+    template_name = "editor/main.html"
+
+    @property
+    def pootle_path(self):
+        return self.object.pootle_path
+
+    @property
+    def ctx_path(self):
+        return self.pootle_path
+
+    @property
+    def vfolder_pk(self):
+        return ""
+
+    @property
+    def project(self):
+        return None
+
+    @property
+    def language(self):
+        return None
+
+    @property
+    def display_vfolder_priority(self):
+        if 'virtualfolder' not in settings.INSTALLED_APPS:
+            return False
+        return VirtualFolderTreeItem.objects.filter(
+            pootle_path__startswith=self.object.pootle_path).exists()
+
+    @property
+    def translate_url(self):
+        return reverse(
+            self.translate_url_path,
+            kwargs=self.url_kwargs)
+
+    @property
+    def browse_url(self):
+        return reverse(
+            self.browse_url_path,
+            kwargs=self.url_kwargs)
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = {}
+        ctx.update(
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Duplicate key 'editor_extends' in dictionary](https://www.codacy.com/app/dwayne/pootle/file/1676475282/issues/source?bid=1990595&fileBranchId=3032697#l493)